packages/react-components/src/helpers/format-date.js

```diff
@@ -0,0 +1,47 @@
+export const isDateAfter = (date1, date2) => {
+  if (!date1 || !(date1 instanceof Date) || !date2 || !(date2 instanceof Date))
+    return;
+
+  return date1.getTime() > date2.getTime();
+};
+
+export const isDateBefore = (date1, date2) => {
+  if (!date1 || !(date1 instanceof Date) || !date2 || !(date2 instanceof Date))
+    return;
+
+  return date1.getTime() < date2.getTime();
+};
+
+export const isDateSameDay = (date1, date2) => {
+  if (!date1 || !(date1 instanceof Date) || !date2 || !(date2 instanceof Date))
+    return;
+
+  return (
+    date1.getFullYear() === date2.getFullYear() &&
+    date1.getMonth() === date2.getMonth() &&
+    date1.getDay() === date2.getDay()
+  );
+};
+
+export const formatDate = (date, pattern, timezone) => {
```

Comment:
Could we do this and not have to use an `||` operator down below?

```suggestion
export const formatDate = (date, pattern, timeZone = 'America/New_York') => {
```